src/components/themed/EdgeOutlinedField.js

```diff
@@ -22,25 +22,36 @@ const ANIMATION_STATES = {
   ERROR: 2
 }
 
-type InputOutlineProps = {
+type Props = {|
+  // Content options:
   label?: string,
   error?: string,
 
+  // Appearance:
+  isClearable?: boolean,
   marginRem?: number | number[],
-  isClearable: boolean,
-  size?: 'big' | 'small',
   showSearchIcon?: boolean,
-  onClear: () => void,
-
-  ...TextInputProps
-}
+  size?: 'big' | 'small',
 
-type InputOutline = {
-  focus: () => void,
-  blur: () => void,
-  isFocused: () => boolean,
-  clear: () => void
-}
+  // Callbacks:
+  onBlur?: () => void,
+  onChangeText?: (text: string) => void,
+  onClear?: () => void,
+  onFocus?: () => void,
+
+  // Other React Native TextInput properties:
+  autoCapitalize?: 'none' | 'sentences' | 'words' | 'characters',
+  autoCorrect?: boolean,
+  autoFocus?: boolean,
+  inputAccessoryViewID?: string,
+  keyboardType?: 'default' | 'number-pad' | 'decimal-pad' | 'numeric' | 'email-address' | 'phone-pad',
+  maxLength?: number,
+  onSubmitEditing?: () => void,
+  returnKeyType?: 'done' | 'go' | 'next' | 'search' | 'send',
+  secureTextEntry?: boolean,
+  testID?: string,
+  value?: string
+|}
```

Comment:
Shouldn't we export the `Props` type from `EdgeTextFieldOutlined` for reuse? Seems like there is a lot of code dup here, but there my be a good reason for it that I am not seeing.

Reply:
Components should never export `Props`. Doing that is a big red flag.

The reason we have to manually put in the `TextInput` properties is because we don't have working Flow types for the `react-native` package, so everything there is just `any`. If we were to switch to typescript, we could just do `Pick<TextInputProps, 'autoCapitalize' | 'autoCorrect' | ...>` to avoid the duplication. I do want to restrict which `TextInput` props are valid, since we don't want anybody passing `style`, for instance.

Reply:
Oh, I saw the comment in the other repo. Yes, the idea is to eventually share the entire component, moving it to another library called `edge-components`. It will be a while before we get there.